official/vision/beta/projects/yolo/modeling/layers/detection_generator.py

```diff
@@ -0,0 +1,275 @@
+"""Contains common building blocks for yolo neural networks."""
+import tensorflow as tf
+import tensorflow.keras as ks
+import tensorflow.keras.backend as K
+
+from official.vision.beta.projects.yolo.ops import box_ops
+
+
+@ks.utils.register_keras_serializable(package='yolo')
+class YoloLayer(ks.Model):
```

Comment:
Is that a yolo model or layer? can you subclass tf.keras.Model or tf.keras.Layer directly?

Reply:
it will only work when built as a keras model, but the name in the paper is the "yolo-layer" so we followed their convention